Assets/Gothic-UnZENity-Core/Scripts/Resource.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using UnityEngine;
+using ZenKit;
+
+namespace GUZ.Core
+{
+	public class Resource<T>
```

Comment:
I like the way you handled this Typed loading properly.
Please add a class comment onto this class and the ResourceLoader class to explain how to use. In the end we normally don't need to understand the details of this magic, but with a comment on both classes, you help people on "howto use it.